### PR TITLE
Slim down the LLVM builds

### PR DESCRIPTION
**Pull request by steven-johnson**
Disable lots of tools and features that we don't need for Halide buildbots. On my Mac, this drops the install dir from ~6GB to ~2GB, and the compressed size from ~1.5GB to ~0.5GB.

### CODE REVIEW

**Review by alexreinking — APPROVED** [review]
I love how each and every one of these options is documented by the LLVM team.

File: master/master.cfg [hunk 1]
```diff
@@ -722,31 +722,174 @@ def get_cmake_build_command(builder_type, build_dir, targets=None):
...
+    "CLANG_TOOL_CLANG_SHLIB_BUILD"
+    "CLANG_TOOL_DIAGTOOL_BUILD"
+    "CLANG_TOOL_DICTIONARY_BUILD"
+    "CLANG_TOOL_LIBCLANG_BUILD"
+    "CLANG_TOOL_NVPTX_ARCH_BUILD"
+    "CLANG_TOOL_SCAN_BUILD_BUILD"
+    "CLANG_TOOL_SCAN_BUILD_PY_BUILD"
+    "CLANG_TOOL_SCAN_VIEW_BUILD"
+    "CLANG_TOOLS_EXTRA_INCLUDE_DOCS"
+    "LLVM_BUILD_32_BITS"
```
Comment (alexreinking) [inline master/master.cfg:763]: Actually, this shouldn't be here. Either that or the part that conditionally sets it should be removed.

**Review by alexreinking — CHANGES_REQUESTED** [review]

**Review by alexreinking — APPROVED** [review]

File: master/master.cfg [hunk 2]
```diff
@@ -722,31 +722,173 @@ def get_cmake_build_command(builder_type, build_dir, targets=None):
     return cmd
 
 
+# options in LLVM that we always want OFF because they aren't needed
+# for Halide buildbot usage.
```
Comment (abadams) [inline master/master.cfg:726]: This comment is wrong and misleading. This doesn't just affect the LLVM used on the buildbots. This is also the LLVM that gets bundled into our libHalide binary releases. Not building the tools has no effect on that, but these decisions are decisions we're making for downstream users of libHalide, not just on our buildbots.

File: master/master.cfg [hunk 3]
```diff
@@ -722,31 +722,173 @@ def get_cmake_build_command(builder_type, build_dir, targets=None):
...
+    "CLANG_TOOL_DIAGTOOL_BUILD"
+    "CLANG_TOOL_DICTIONARY_BUILD"
+    "CLANG_TOOL_LIBCLANG_BUILD"
+    "CLANG_TOOL_NVPTX_ARCH_BUILD"
+    "CLANG_TOOL_SCAN_BUILD_BUILD"
+    "CLANG_TOOL_SCAN_BUILD_PY_BUILD"
+    "CLANG_TOOL_SCAN_VIEW_BUILD"
+    "CLANG_TOOLS_EXTRA_INCLUDE_DOCS"
+    "LLVM_BUILD_UTILS"
+    "LLVM_ENABLE_BINDINGS"
```
Comment (abadams) [inline master/master.cfg:764]: This looks like the same set of features of libLLVM that were disabled before the PR, so my beef is strictly with the comment.

File: master/master.cfg (hunk 2 above)
Comment (steven-johnson) [reply master/master.cfg:726]: I'm going to fix it in https://github.com/halide/build_bot/pull/276

**Pull request merged**